### ISSUE TITLE
annotation documentation for middleware incorrect

### ISSUE THREAD

**Issue by travisghansen** [containous/traefik#6506]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Try adding a middleware to an `Ingress` per the doc of the new feature in 2.2.

https://github.com/containous/traefik/pull/6160/files

```
traefik.ingress.kubernetes.io/router.middlewares: prefix@kuberntes-crd
```

### What did you expect to see?

The middleware applied to the `Ingress`

### What did you see instead?

Failure and a `404` on the endpoint.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2.2.0-rc2
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

`Ingress` mode installed via the helm chart.

Simply removing the `-` from the `namespace` aspect of the middleware works.

ie: `traefik.ingress.kubernetes.io/router.middlewares: prefix@kuberntescrd`

Also, it may be good to give some explanation about the autogenerated names of the middlewares created via `CRD`. For example explain something like:

```
Middlewares created via CRD take the format of `<k8s namespace>-<k8s name>@kubernetescrd`.
```

**Comment by ldez** [comment]
Hello,

could you open a PR on the branch v2.2 to fix that?

Thanks.

**Comment by travisghansen** [comment]
I'm pretty strapped for time currently. Mostly just letting you know. If I get a chance I'll see what I can do though.

**Comment by ldez** [comment]
Fixed in d5cbb6028ff72f197b74f8f6e47f7c069058426c

**Comment by traefiker** [comment]
Closed by #6504.
